### ISSUE TITLE
[Candlelight] Marriage!

### ISSUE THREAD

**Issue by norfaremc** [satisfyu/Let-s-Do-Hub#368]
### Currently no capacity for new suggestions.

- [X] I understand that my suggestion might be deleted.

### Mod Name and Version

Candlelight

### Suggestion Details

Hello! I love this mod, and it's perfect for adding a romantic RPG aspect to my server. I feel like the only thing this mod is missing at this point is.... marriage!

some of my suggestions:
- A way to propose to players and marry them with the ring/ringbox from the mod would be so darn cute and my server players would love it! We have many irl couples on our server playing together, and being able to recognize their relationship in-game, in a fun way, would be great for them. Many players have requested it from me.
- Their tamed animals are your tamed animals too. Ever had a tamed animal in the way of your building while your partner is offline? That's fine, you can move them! However, the main owner of every pet is, when online, the person that the animal follows. However, when they are not online, it follows you.
- Health bar indicator for your spouse, so no matter where they are, you know how they're doing.
- Distance from partner indicator (under their health bar), or direct coordinates so you always know where they are. Useful when traveling, or, when you and you partner are stuck in a cave and your partner is "in trouble", you know where to find them.
maybe have a chat indicator of marriage [Married to: (username)] The color of the marriage tag would change based on your Marriage Happiness (see below)
- Cute ways you can show your spouse love in-game, like crouching and "smoochin" your partner on the cheek (heart display like when you tame animals maybe?)
-A marriage happiness indicator in-game that you have to work on to maintain a certain status, like gifting your partner letters, chocolate boxes, playing a certain amount of hours together, etc. Essentially players have to work up their happiness bar to get special chat titles, or be able to do things like "smooch" them, tp to them, etc.
- A marriage ceremony with a priest to solidify the marriage. Vanilla mc priest villager can be an option to purchase a marriage license and have a "ceremony". People could also plan their ceremonies and exchange rings in front of their friends.
- The mod already has a dress and a suit, but adding a wedding dress and tux would be nice.
- Being able to PM them without having to type out the long dreaded /msg (name) (message), it could be /msgspouse, and it could default the chat to auto-type out the command every time you send a message, so you don't have to keep typing it, until you delete the auto command and then it defaults back to server messaging.
- Unique, customizable titles above player names in chat or above heads, such as "[Name]'s Beloved" or "Partner of [Name]."
- Unlock unique abilities when both players are doing things together like mining - increased mining speed, swimming - faster swimming or farming - faster crop regeneration.
- Notification when your partner is sleeping.
- Two person mount for horses and mules? 
- The longer a couple stays married, the stronger their buffs become (e.g., increased XP gain, etc).
- On special in-game anniversaries, fireworks or confetti effects trigger around the couple.
- A book item where couples and friends can save milestones, screenshots, or short written notes about their adventures. 
- Divorce process would be costly. You'd have to buy divorce papers. Both players don't have to agree to divorce to end a marriage, but once ended, you have to rebuild the Marriage Happiness if you were to get married again.
- Marriage cooldown--players can't just marry anyone, divorce, and start over and over again.
- Being able to toggle PvP for couples in PvP servers, so they don't accidentally hit their spouse.
- A way to set their marriage to "public" or "private", so they can choose if they want server chat messages or display PDA.

There's only one mod for forge that adds marriage for players, and it's MCA.... which is just not a mod I'm interested in adding to my modpack. Right now, there's no real reason for players to use the ring box, the clothing, the flowers, letters, etc.

Please consider my idea :), I know it is a lot I'm sorry lol https://www.curseforge.com/minecraft/bukkit-plugins/marriage-master is a good inspiration.

**TLDR; I made a million suggestions, but all of it is just fluff and possibilities. Ultimately, utilizing the items you created in game that don't necessarily serve a real purpose can have a purpose through a marriage system, that is worth player's time.** 

### Expected Benefits

More RPG gameplay and fun for players on private worlds or public servers. 
Platonic friends can utilize the system to buddy-up, utilizing the tamed animals feature, health bar/distance indicators, the private messaging, etc. They can disable PDA and change their marriage to private to utilize the benefits without displaying marriage.

### Check for Similar Suggestions

- [X] Yes, I've looked for similar ideas.

**Comment by satisfyu** [comment]
Thank you for your detailed and creative feedback!

Your ideas for a marriage system are great and would definitely add a lot of charm and depth to the gameplay experience.

However, a feature of this scale would require a significant amount of development work - so much so that it would likely be better suited as an entirely separate mod rather than just an addition to Candlelight. Since im working on the mod alone and already have my hands full with other tasks, i unfortunately don't have the capacity to take on something this extensive at the moment. :/

That being said, I truly appreciate your enthusiasm and will keep your suggestions in mind should the opportunity arise to explore this concept further in the future.

I hope you continue to enjoy Candlelight. :)
